content/articles/rest-api-with-golang-without-any-frameworks-only-using-native-modules/index.md

```diff
@@ -0,0 +1,421 @@
+Go is a fast-growing language among developers. Its popularity is still growing and unshaken. Golang ecosystem has a tone of native libraries. However, it has third-party libraries that developers can still choose to use.
+
+When creating Rest API with Go, you might choose to use libraries such as Gorm, Go fiber, Gin, fast HTTP, etc. On the same, you can still combine this with other third-party libraries to come up with the API of your choice.
+
+However, due to the many libraries that the Go ecosystem has, you can still choose native libraries that you don't have to download and build your APIs. In this guide, we will build a complete and functional REST API. This API will only use native libraries that Golang has.
+
+### Requirements
+
+To follow along with this tutorial, it's essential to have prior knowledge of how to work with Golang. Therefore, ensure you have the Golang IDE installed on your computer to execute the Golang code. Once installed, run the `go version` command to verify Go is installed.
+
+Check this [guide](/engineering-education/golang-part-2-programming-basics/) to get started with how to write and run Golang programs.
+
+### Set a Golang project
+
+To start a Go project, you need to first initialize Go within your project. In this case, create a project folder where you want your app to live. Open a command line that points to the newly created directory, then initialize Golang using the following command:
+
+```bash
+go mod init native-go-api
+```
+
+This will initialize a Go project by creating a `go.mod` file. Any go package that you install, its dependencies are saved here. The `go.mod` file has a `native-go-api` module name that helps you create local modules and import them so that other local modules of your project can use them.
+
+### Create a CRUD Golang REST API using native modules
+
+Let's dive in and create a Golang REST API using native modules. This tutorial will use a movie use case. First, model the movie data you want your API to interact with. To do this, create a `models` folder at the root directory of your project. Then create a `movie.go` file and add the following model:
```

Comment:
Describe the project you are building clearly. Saying the project is a "Movie use case" is not enough.